source/game/kart/KartJump.cc

```diff
@@ -0,0 +1,258 @@
+#include "KartJump.hh"
+
+#include "game/kart/KartCollide.hh"
+#include "game/kart/KartMove.hh"
+#include "game/kart/KartParam.hh"
+#include "game/kart/KartPhysics.hh"
+#include "game/kart/KartState.hh"
+
+#include "game/system/KPadController.hh"
+
+#include <egg/math/Math.hh>
+
+namespace Kart {
+
+KartJump::KartJump(KartMove *move) : m_move(move) {
+    m_cooldown = 0;
+}
+
+KartJump::~KartJump() = default;
+
+void KartJump::setupProperties() {
+    static constexpr std::array<TrickProperties, 3> TRICK_PROPERTIES = {{
+            {11.0f, 1.5f, 0.9f, 0.0018f},
+            {14.0f, 1.5f, 0.9f, 0.0006f},
+            {7.5f, 2.5f, 0.93f, 0.05f},
+    }};
+
+    static constexpr std::array<f32, 3> FINAL_ANGLES = {{
+            360.0f,
+            720.0f,
+            180.0f,
+    }};
+
+    if (m_variant == BoostRampVariant::SingleFlipTrick) {
+        m_properties = TRICK_PROPERTIES[0];
+        m_finalAngle = FINAL_ANGLES[0];
+    } else if (m_variant == BoostRampVariant::DoubleFlipTrick) {
+        m_properties = TRICK_PROPERTIES[1];
+        m_finalAngle = FINAL_ANGLES[1];
+    } else if (m_type == TrickType::BikeSideStuntTrick) {
+        m_properties = TRICK_PROPERTIES[2];
+        m_finalAngle = FINAL_ANGLES[0];
+    }
+
+    m_angleDelta = m_properties.initialAngleDiff;
+    m_angleDeltaFactorDec = m_properties.angleDiffMulDec;
+    m_angle = 0.0f;
+    m_angleDeltaFactor = 1.0f;
+}
+
+void KartJump::reset() {
+    m_cooldown = 0;
+}
+
+void KartJump::tryStart(const EGG::Vector3f &left) {
+    if (!state()->isTrickStart()) {
+        return;
+    }
+
+    if (m_move->speedRatioCapped() > 0.5f) {
+        m_variant = static_cast<BoostRampVariant>(state()->boostRampType());
+
+        start(left);
+    }
+
+    state()->setTrickStart(false);
+}
+
+void KartJump::calc() {
+    --m_cooldown;
+    m_cooldown = std::max(0, static_cast<s32>(m_cooldown));
```

Comment:
```suggestion
    --m_cooldown;
    m_cooldown = std::max(0, static_cast<s32>(m_cooldown - 1));
```
If it uses a `cmpwi` instruction, it's signed.